runs.py

```diff
@@ -120,6 +123,58 @@ def removals(self) -> list[str]:
         all_removals = self.neow_bonus.cards_removed + event_removals + store_removals
         return all_removals
 
+    @property
+    def character_streak(self) -> StreakInfo:
+        """Get the run position in the character streak."""
+        if self._character_streak is None:
+            self._character_streak = self._get_streak(is_character_streak=True)
+        return self._character_streak
+
+    @property
+    def rotating_streak(self) -> StreakInfo:
+        """Get the run position in the rotating streak."""
+        if self._rotating_streak is None:
```

Comment:
this None check is how I'm making sure it only loads one time instead of every time it's called. Similar to the cache d seed on FileParser

Reply:
There is a problem with this though, because we aren't reloading it. When a current run finishes (win) and it's added to the Run History, the prior runs in the streaks are not being updated. We may need some method to reset the streaks so they can be reloaded to include the new run. We should only have to reload the runs up to the most recent loss of the character that just won.

EX: 
1. Silent won, Defect won, Watcher won, current streak is 3
2. Ironclad run is ongoing
3. Ironclad run wins, its streak on Run history says 4/4 (4th run in the streak of 4)
4. The 3 runs before that will still say 1/3, 2/3, 3/3.
5. We should only have to reset the `StreakInfo` of any runs since the last Ironclad loss. I believe this should always work for the logic we're going with.

I'm just not sure where it needs to be reset, presumably when the save becomes a run file and is added to History, but I'm not familiar enough with the code to know where that happens